detectors/gcp/gce.go

```diff
@@ -39,10 +39,25 @@ func (d *Detector) GCEHostID() (string, error) {
 }
 
 // GCEHostName returns the instance name of the instance on which this program is running.
+// Recommended to use GCEInstanceName() or GCEInstanceHostname() to more accurately reflect which
+// value is returned.
```

Comment:
I thought about that, wasn't sure if we could since this is stable. 

Though, `GCEHostName()` maps to the `host.name` attribute (while `GCEInstanceName()`->`gce.instance.name` and `GCEInstanceHostname()` -> `gce.instance.hostname`). When we added the new attributes, it was suggested that upstream might want to add something like a `host.dns_name` to cover the semantics of `gce.instance.hostname`. 

In that case, we would want to deprecate the *new* attributes instead. So I think holding off on choosing which one to deprecate is best for now, but I agree at some point we should drop one of them.

Reply:
Sounds good.